pkg/controller/storagecluster/storagecluster.go

```diff
@@ -405,6 +406,54 @@ func (c *Controller) validateCloudStorageLabelKey(cluster *corev1.StorageCluster
 	return nil
 }
 
+func (c *Controller) runPreflightCheck(cluster *corev1.StorageCluster) error {
+	check, ok := cluster.Annotations[pxutil.AnnotationPreflightCheck]
+	if !ok || check == "skip" {
+		return nil
+	} else if check == "false" {
+		if cluster.Status.Phase == string(corev1.ClusterConditionTypePreflight)+string(corev1.ClusterOperationFailed) {
```

Comment:
define new phase for it, the concatenated string is not in good format, also it's not easy to refer in multiple places.

Reply:
will use condition instead of phase, will see how I can address this there.